src/libraries/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.ParseMapModules.cs

```diff
@@ -0,0 +1,146 @@
+                else if (s[i] == 'x')
+                    sawExec = true;
+            }
+
+            return sawRead & sawExec;
+        }
+
+        internal static ProcessModuleCollection? ParseMapsModules(int pid)
+        {
+            try
+            {
+                return ParseMapsModulesCore(File.ReadLines(GetMapsFilePathForProcess(pid)));
+            }
+            catch (IOException) { }
+            catch (UnauthorizedAccessException) { }
+
+            return null;
+        }
+
+        private static ProcessModuleCollection ParseMapsModulesCore(IEnumerable<string> lines)
+        {
+            Debug.Assert(lines != null);
+
+            ProcessModule? module = null;
+            ProcessModuleCollection modules = new(capacity: 0);
+            bool moduleHasReadAndExecFlags = false;
+
+            foreach (string line in lines)
+            {
+                // Use a StringParser to avoid string.Split costs
+                var parser = new StringParser(line, separator: ' ', skipEmpty: true);
+
+                // Parse the address start and size
+                (long Start, int Size) addressRange = parser.ParseRaw(TryParseAddressRange);
+
+                if (addressRange == default)
+                {
+                    Commit();
+                    continue;
+                }
+
+                // Parse the permissions
+                bool lineHasReadAndExecFlags = parser.ParseRaw(HasReadAndExecFlags);
+
+                // Skip past the offset, dev, and inode fields
+                parser.MoveNext();
+                parser.MoveNext();
+                parser.MoveNext();
+
+                // we only care about the named modules
+                if (!parser.MoveNext())
+                {
+                    Commit();
+                    continue;
+                }
+
+                // Parse the pathname
+                string pathname = parser.ExtractCurrentToEnd();
+                bool isContinuation = module?.FileName == pathname && (long)module.BaseAddress + module.ModuleMemorySize == addressRange.Start;
+
+                if (isContinuation)
+                {
+                    module!.ModuleMemorySize += addressRange.Size;
+                    moduleHasReadAndExecFlags |= lineHasReadAndExecFlags;
+                    continue;
+                }
+
+                Commit();
+
+                moduleHasReadAndExecFlags = lineHasReadAndExecFlags;
+                module = new ProcessModule
+                {
+                    FileName = pathname,
+                    ModuleName = Path.GetFileName(pathname),
+                    ModuleMemorySize = addressRange.Size,
+                    EntryPointAddress = IntPtr.Zero // unknown
+                };
+
+                // on 32-bit platforms, it throws System.OverflowException without the void* cast.
+                unsafe
+                {
+                    module.BaseAddress = new IntPtr(unchecked((void*)addressRange.Start));
+                }
```

Comment:
```suggestion
                // on 32-bit platforms, it throws System.OverflowException with IntPtr.ctor(Int64),
                // so we use IntPtr.ctor(void*) to skip the overflow checking.
                unsafe
                {
                    module.BaseAddress = new IntPtr((void*)addressRange.Start);
                }
```
based on https://github.com/dotnet/runtime/blob/34cae7461f96a9c1a7574c72c436e309bf4031f4/src/libraries/System.Private.CoreLib/src/System/IntPtr.cs#L40-L55 and https://github.com/dotnet/runtime/pull/45480#discussion_r534255927?